src/Bicep.Core/Analyzers/Linter/Rules/SecureParameterDefaultRule.cs

```diff
@@ -34,6 +34,11 @@ override public IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel model, Di
                     // Empty string - okay
                     continue;
                 }
+                else if (defaultValue is VariableAccessSyntax variableAccessSyntax)
+                {
+                    // Variable - okay
+                    continue;
+                }
```

Comment:
Could this check be type-driven instead? Something like `else if (model.GetTypeInfo(defaultValue).ValidationFlags.HasFlag(TypeSymbolValidationFlags.IsSecure)) ...`?

Reply:
Sure will give this a shot. I did think the fix was too simplistic.

Reply:
@jeskew have amended as per suggestion, and added more unit tests

Reply:
Amended as per suggestion @jeskew